extensions/ql-vscode/test/unit-tests/model-editor/modeled-method.test.ts

```diff
@@ -50,3 +54,27 @@ describe("modeledMethodSupportsProvenance", () => {
     }
   });
 });
+
+describe("createModeledMethodKey", () => {
+  it("should create a key for a modeled method", () => {
+    const modeledMethod = createNoneModeledMethod();
+    const key = createModeledMethodKey(modeledMethod);
+
+    const expectedKey =
+      '{"endpointType":"method","methodName":"createQuery","methodParameters":"(String)","packageName":"org.sql2o","signature":"org.sql2o.Connection#createQuery(String)","type":"none","typeName":"Connection"}';
+
+    expect(key).toBe(expectedKey);
+  });
+
+  it("should check that two modeled methods are the same", () => {
+    const modeledMethod = createSummaryModeledMethod();
+    const key = createModeledMethodKey(modeledMethod);
+
+    const modeledMethod2 = createSummaryModeledMethod();
+    const key2 = createModeledMethodKey(modeledMethod2);
+
+    // Object references are different, but the keys are the same.
+    expect(modeledMethod === modeledMethod2).toBe(false);
+    expect(key === key2).toBe(true);
```

Comment:
This should result in better error messages:

```suggestion
    expect(modeledMethod).not.toBe(modeledMethod2);
    expect(key).toEqual(key2);
```